File: jvm/sargon-android/src/main/java/com/radixdlt/sargon/extensions/ResourceIndicator.kt
package com.radixdlt.sargon.extensions

import com.radixdlt.sargon.ResourceAddress
import com.radixdlt.sargon.ResourceIndicator
import com.radixdlt.sargon.resourceIndicatorGetAddress

val ResourceIndicator.address: ResourceAddress
    get() = resourceIndicatorGetAddress(indicator = this)

File: crates/sargon-uniffi/src/wrapped_radix_engine_toolkit/low_level/execution_summary/resource_indicator/resource_indicator.rs
use crate::prelude::*;
use sargon::ResourceIndicator as InternalResourceIndicator;

#[derive(Clone, PartialEq, Eq, InternalConversion, uniffi::Enum)]
pub enum ResourceIndicator {
    Fungible {
        resource_address: ResourceAddress,
        indicator: FungibleResourceIndicator,
    },
    NonFungible {
        resource_address: ResourceAddress,
        indicator: NonFungibleResourceIndicator,
    },
}

#[uniffi::export]
pub fn new_resource_indicator_sample() -> ResourceIndicator {
    InternalResourceIndicator::sample().into()
}

#[uniffi::export]
pub fn new_resource_indicator_sample_other() -> ResourceIndicator {
    InternalResourceIndicator::sample_other().into()
}

#[uniffi::export]
pub fn resource_indicator_get_address(
    indicator: &ResourceIndicator,
) -> ResourceAddress {
    indicator.into_internal().get_address().into()
}
